src/hb-font.cc

```diff
@@ -2067,13 +2067,16 @@ hb_font_set_variations (hb_font_t            *font,
   const OT::fvar &fvar = *font->face->table.fvar;
   for (unsigned int i = 0; i < variations_length; i++)
   {
-    unsigned axis_index;
-    if (fvar.find_axis_index (variations[i].tag, &axis_index) &&
-	axis_index < coords_length)
+    unsigned axis_index_start, axis_index_end;
+    if (fvar.find_axis_index (variations[i].tag, &axis_index_start, &axis_index_end))
```

Comment:
If there are multiple axes with the same tag, is it ensured that they are ordered and identical tags are lined up consecutively? In the font, or does HB sort them somewhere?

Reply:
PS: If yes, then LGTM.

Reply:
I don't think the OT spec demands the axes to be ordered by tag, and `find_axis_index ` seems to iterate using the order in the font: I suspect that `find_axis_index()` is not correct.

Reply:
Ah, I assumed they are sorted by tag. Let me check.

Reply:
Okay yes, not sorted. I'll rework the code.

Reply:
I pushed a reworked version that doesn't make sorted assumption. PTAL.

Reply:
Looks good to me.